lib/chromedriver.js

```diff
@@ -254,53 +259,109 @@ class Chromedriver extends events.EventEmitter {
+        return executable;
+      }
 
-    if (!chromeVersion) {
-      // unable to get the chrome version
-      let cd = cds[0];
-      log.warn(`Unable to discover Chrome version. Using Chromedriver ${cd.version} at '${cd.executable}'`);
-      return cd.executable;
-    }
+      const chromeVersion = await this.getChromeVersion();
+      if (!chromeVersion) {
+        // unable to get the chrome version
+        const {version, executable} = cds[0];
+        log.warn(`Unable to discover Chrome version. Using Chromedriver ${version} at '${executable}'`);
+        return executable;
+      }
 
-    log.debug(`Found Chrome bundle '${this.bundleId}' version '${chromeVersion}'`);
-
-    if (semver.gt(chromeVersion, _.values(mapping)[0]) &&
-        !_.isUndefined(cds[0]) && _.isUndefined(cds[0].minChromeVersion)) {
-      // this is a chrome above the latest version we know about,
-      // and we have a chromedriver that is beyond what we know,
-      // so use the most recent chromedriver that we found
-      let cd = cds[0];
-      log.warn(`No known Chromedriver available to automate Chrome version '${chromeVersion}'.\n` +
-               `Using Chromedriver version '${cd.version}', which has not been tested with Appium.`);
-      return cd.executable;
-    }
+      log.debug(`Found Chrome bundle '${this.bundleId}' version '${chromeVersion}'`);
 
-    const workingCds = cds.filter((cd) => {
-      return !_.isUndefined(cd.minChromeVersion) && semver.gte(chromeVersion, cd.minChromeVersion);
-    });
+      const autodownloadMsg = this.storageClient && didStorageSync
+        ? ''
+        : '. You could also try to enable automated chromedrivers download server feature';
+      if (semver.gt(chromeVersion, _.values(mapping)[0]) && cds[0] && !cds[0].minChromeVersion) {
+        if (this.storageClient && !didStorageSync) {
+          try {
+            if (await syncChromedrivers(chromeVersion)) {
+              continue;
+            }
+          } catch (e) {
+            log.warn(e.stack);
+          }
+        }
+        // this is a chrome above the latest version we know about,
+        // and we have a chromedriver that is beyond what we know,
+        // so use the most recent chromedriver that we found
+        const {version, executable} = cds[0];
+        log.warn(`No known Chromedriver available to automate Chrome version '${chromeVersion}'.\n` +
+          `Using Chromedriver version '${version}', which has not been tested with Appium` +
+          autodownloadMsg);
+        return executable;
+      }
 
-    if (_.isEmpty(workingCds)) {
-      log.errorAndThrow(`No Chromedriver found that can automate Chrome '${chromeVersion}'. ` +
-                        `See ${CHROMEDRIVER_TUTORIAL} for more details.`);
-    }
+      const workingCds = cds.filter((cd) => {
+        const versionObj = semver.coerce(cd.minChromeVersion);
+        return versionObj && chromeVersion.major === versionObj.major;
```

Comment:
why use ”==“ ，the chromedriver ability to support multiple versions,A range should be set
![image](https://user-images.githubusercontent.com/31992251/67938843-3f660080-fc0b-11e9-976d-e34828f3eca7.png)

Reply:
Since Chrome 73 the versions of Chrome and Chromedriver have been aligned. Chrome 74.x.x requires Chromedriver 74.x.x. Chrome 76.x.x requires Chromedriver 76.x.x